lib/devices/light.js

```diff
@@ -27,103 +31,144 @@ module.exports = function(device, api, capabilities) {
-    .on('set', function(value, callback) {
-      device.setCapabilityValue('onoff', value)
-      callback();
-    })
-    .on('get', returnCapabilityValue(device, 'onoff'));
-
-  //Brightness
-  if ('dim' in capabilities) {
-    homekitAccessory
-      .getService(Service.Lightbulb)
+  if (capabilityBaseName === 'onoff') {
+    service
+      .getCharacteristic(Characteristic.On)
+      .on('set', function(value, callback) {
+        device.setCapabilityValue(capability, value)
+        callback();
+      })
+      .on('get', returnCapabilityValue(device, capability));
+  }
+
+  // Brightness
+  if (capabilityBaseName === 'dim') {
+    service
       .addCharacteristic(Characteristic.Brightness)
       .on('set', debounce(function(value, callback) {
-        device.setCapabilityValue('dim', value / 100)
+        device.setCapabilityValue(capability, value / 100)
         callback();
       }, 500))
-      .on('get', returnCapabilityValue(device, 'dim', v => v * 100));
+      .on('get', returnCapabilityValue(device, capability, v => v * 100));
   }
 
-  //Saturation
-  if ('light_saturation' in capabilities) {
-    homekitAccessory
-      .getService(Service.Lightbulb)
+  // Saturation
+  if (capabilityBaseName === 'light_saturation') {
+    service
       .addCharacteristic(Characteristic.Saturation)
       .on('set', function(value, callback) {
-        device.setCapabilityValue('light_saturation', value / 100)
+        device.setCapabilityValue(capability, value / 100)
         callback();
       })
-      .on('get', returnCapabilityValue(device, 'light_saturation', v => v * 100));
+      .on('get', returnCapabilityValue(device, capability, v => v * 100));
   }
 
-  //Temperature
-  if ('light_temperature' in capabilities) {
-    homekitAccessory
-      .getService(Service.Lightbulb)
+  // Temperature
+  if (capabilityBaseName === 'light_temperature') {
+    service
       .addCharacteristic(Characteristic.ColorTemperature)
       .on('set', function(value, callback) {
 
-    // If device uses light_mode and was previously set on not temperature
-    // wait for color mode to be adjusted before setting light_temperature
-    if ('light_mode' in (device.capabilitiesObj || {}) && device.capabilitiesObj.light_mode.value !== 'temperature') {
-      device.setCapabilityValue('light_mode', 'temperature');
-    }
-
-    device.setCapabilityValue('light_temperature', map(140, 500, 0, 1, value))
-      callback();
-    }).on('get', function(callback) {
-      const value = 'light_temperature' in (device.capabilitiesObj || {}) ? device.capabilitiesObj.light_temperature.value : 1;
-      callback(null, map(0, 1, 140, 500, value) || 500);
-    });
+        // If device uses light_mode and was previously set on not temperature
+        // Wait for color mode to be adjusted before setting light_temperature
+        if ('light_mode' in (device.capabilitiesObj || {}) && device.capabilitiesObj.light_mode.value !== 'temperature') {
```

Comment:
To complicate matters, if a device has multiple `light_temperature` capabilities, it may also have multiple (matching) `light_mode` capabilities 🥴

But I'm fine with leaving this as-is, because it's unlikely that it will have different separate modes for multiple lights.